### DISCUSSION TITLE
ReactPHP without Nginx?

### DISCUSSION THREAD

**Discussion by michaelphipps** [reactphp/reactphp#470] [Q&A] [answered]
What's the best approach for pointing sub domains at reactphp/http servers on the same ip?  Nginx?  Or can I do something funky with reactphp?  The usecase is breaking things into microservices:  Say I have 3 API servers running  on the same machine.  I want to call these servers from a javascript client via https://auth.domain.name, https://api.domain.name, https://lookup.domain.name.  I know I can set up Nginx to do this, but I want to understand if this is the best approach, or if there would be an advantage using another ReactPHP server as a gateway/proxy?

**Comment by SimonFrings** [comment, marked as answer]
Hey @michaelphipps, I think in this case I would recommend using something like nginx for this. It is possible using the ReactPHP server, but you should be careful what requests to accept . You have to set up the whole security/encryption part for this server by yourself, which is not an easy task to do and could lead to security issues for your microservices. I guess you could run this behind a reverse proxy, but I personally would go with the easy option here.

Nginx for example does all this already, that's why I'm recommending it. 👍

**Reply by clue** [reply]
I agree that using a reverse proxy such as nginx or HAProxy is a good idea if you want to run multiple applications on the same host 👍 It's pretty common to set this up to also take care of secure HTTPS and only forward unencrypted HTTP to your backend server processes.

> it is possible using the ReactPHP server, but you should not put this server on the internet (bad description but it fits my point ^^).

ReactPHP can be (and is) used on the Internet, so it's definitely safe to use here. The thing is, once you open up for an unknown audience, you want to be careful about what requests you accept. As such, it's usually a good idea to stick to something well established such as nginx to avoid many of the unknown unknowns. Better be safe than sorry 👍

**Reply by SimonFrings** [reply]
@clue Thanks for clarifying, I wanted to say the same thing but it came out a bit wrong 😅

**Reply by michaelphipps** [reply]
Thanks for the helpful responses!

**Reply by clue** [reply]
@arunnabraham I don't think nghttp2/nghttp3 is related to this question or answer, but if there's anything you want to discuss, perhaps create a separate question or answer?

**Reply by arunnabraham** [reply]
@clue Sorry to put a statement here. I can put a thread for this. Updating..

**Comment by quazardous** [comment]
I'm running 10 reactphp/http wrapped in SF command + supervisor with HAProxy on top

2k req/s no sweat ;p

haproxy is handling the SSH termination so behind it must be safe (like a DMZ).

I used nginx at first but there was some buffering issue with the public branch of nginx (too much magic).

**Reply by SimonFrings** [reply]
> 2k req/s no sweat ;p

Sound like a nice ReactPHP testimonial :D
